File: dataset/human/c/file_1660.c
#include <stdio.h>

int main()
{
    int n;
    scanf("%d",&n);
    int a[30];
    int b[30];
    int c[30];
    for(int i=0;i<n;i++) scanf("%d",&a[i]);
    for(int i=0;i<n;i++) scanf("%d",&b[i]);
    for(int i=0;i<n-1;i++) scanf("%d",&c[i]);

    int  ans=0;
    ans+=b[a[0]-1];
    for(int i=1;i<n;i++)
    {
        ans+=b[a[i]-1];
        if(a[i]==a[i-1]+1)
        {
            ans+=c[a[i-1]-1];
        }
    }
    printf("%d\n",ans);
}

File: dataset/human/java/file_1660.java
import java.util.Scanner;

public class Main {

	static long MOD = 1000000007;

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int N = in.nextInt();
		char[] S = in.next().toCharArray();
		int[] alph = new int[26];
		for (int i = 0; i < S.length; i++) {
			alph[S[i] - 'a']++;
		}
		long ans = 1;
		for (int i = 0; i < alph.length; i++) {
			if (alph[i] > 0) {
				int num = alph[i] + 1;
				ans = ans * num % MOD;
			}
		}
		System.out.println(ans - 1);
		in.close();
	}
}
